lib/main.coffee

```diff
@@ -49,14 +54,17 @@ module.exports =
         filePath = textEditor.getPath()
         command = @executablePath
         ruleset = @rulesets
-        if @projectRules
-          rulesetPath = helpers.find(filePath, 'ruleset.xml')
+        if /^[a-z0-9]+\.xml$/gi.test(@rulesets)
+          rulesetPath = helpers.find(filePath, @rulesets)
           ruleset = rulesetPath if rulesetPath?
         parameters = []
         parameters.push(filePath)
         parameters.push('text')
         parameters.push(ruleset)
-        return helpers.exec(command, parameters).then (output) ->
+        options = {}
+        projectDir = atom.project.relativizePath(filePath)[0]
+        options.cwd = projectDir or textEditor.getDirectoryPath()
```

Comment:
I can't find this `getDirectoryPath()` function in [the documentation](https://atom.io/docs/api/v1.7.4/TextEditor). Are you sure it's part of the public API? If so what object is it really?

All the other linters are using `path.dirname(filePath)` (with `filePath` set via `textEditor.getPath()`).